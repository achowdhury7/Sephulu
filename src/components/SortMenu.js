import React, { Component } from 'react'
import { Nav, NavItem } from 'react-bootstrap'

class SortMenu extends Component {
  constructor(props) {
    super(props)
    this.state = {
      activeKey: ''
    }
    this.handleSelect = this.handleSelect.bind(this)
  }

  handleSelect(key) {
    if (this.state.activeKey !== key) {
      this.setState({
        activeKey: key
      })
  
    if (this.props.handleSelect) this.props.handleSelect(key)
    }
  }

  render() {
    const menuItems = Object.keys(this.props.menuItems).map(key => (
        <NavItem className="sort-menu-item" key={key} eventKey={key} href="#">{this.props.menuItems[key]}</NavItem>
      )
    )

    return (
      <div className="sort-menu-container">
        <span>Sort By:</span>
        <span>
          <Nav bsStyle="pills" activeKey={this.state.activeKey} className="sort-menu" onSelect={this.handleSelect}>
            {menuItems}
          </Nav>
        </span>
      </div>
    )
  }
}

export default SortMenu