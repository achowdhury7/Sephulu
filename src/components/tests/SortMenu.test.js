import React from 'react'
import { shallow, mount, render } from 'enzyme'

import SortMenu from '../SortMenu'

let wrapper

beforeEach(() => {
  const menuItems = ['a', 'b', 'c']
  wrapper = mount(
    <SortMenu menuItems={menuItems} />
  )
})

test('render a nav menu', () => {
  expect(wrapper).toMatchSnapshot()
})

test('set `activeKey` on menu item click', () => {
  const firstChildKey = wrapper.find('NavItem').first().prop('eventKey')

  expect(wrapper).toMatchSnapshot()

  wrapper.find('a').first().simulate('click')
  expect(wrapper.state('activeKey')).toEqual(firstChildKey)
})
