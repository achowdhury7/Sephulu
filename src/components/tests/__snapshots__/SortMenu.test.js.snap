// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render a nav menu 1`] = `
<SortMenu
  menuItems={
    Array [
      "a",
      "b",
      "c",
    ]
  }
>
  <div
    className="sort-menu-container"
  >
    <span>
      Sort By:
    </span>
    <span>
      <Nav
        activeKey=""
        bsClass="nav"
        bsStyle="pills"
        className="sort-menu"
        justified={false}
        onSelect={[Function]}
        pullLeft={false}
        pullRight={false}
        stacked={false}
      >
        <ul
          className="sort-menu nav nav-pills"
          role={null}
        >
          <NavItem
            active={false}
            activeKey=""
            className="sort-menu-item"
            disabled={false}
            eventKey="0"
            href="#"
            key=".$0"
            onSelect={[Function]}
          >
            <li
              className="sort-menu-item"
              role="presentation"
            >
              <SafeAnchor
                componentClass="a"
                disabled={false}
                href="#"
                onClick={[Function]}
                role="button"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  role="button"
                >
                  a
                </a>
              </SafeAnchor>
            </li>
          </NavItem>
          <NavItem
            active={false}
            activeKey=""
            className="sort-menu-item"
            disabled={false}
            eventKey="1"
            href="#"
            key=".$1"
            onSelect={[Function]}
          >
            <li
              className="sort-menu-item"
              role="presentation"
            >
              <SafeAnchor
                componentClass="a"
                disabled={false}
                href="#"
                onClick={[Function]}
                role="button"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  role="button"
                >
                  b
                </a>
              </SafeAnchor>
            </li>
          </NavItem>
          <NavItem
            active={false}
            activeKey=""
            className="sort-menu-item"
            disabled={false}
            eventKey="2"
            href="#"
            key=".$2"
            onSelect={[Function]}
          >
            <li
              className="sort-menu-item"
              role="presentation"
            >
              <SafeAnchor
                componentClass="a"
                disabled={false}
                href="#"
                onClick={[Function]}
                role="button"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  role="button"
                >
                  c
                </a>
              </SafeAnchor>
            </li>
          </NavItem>
        </ul>
      </Nav>
    </span>
  </div>
</SortMenu>
`;

exports[`set \`activeKey\` on menu item click 1`] = `
<SortMenu
  menuItems={
    Array [
      "a",
      "b",
      "c",
    ]
  }
>
  <div
    className="sort-menu-container"
  >
    <span>
      Sort By:
    </span>
    <span>
      <Nav
        activeKey=""
        bsClass="nav"
        bsStyle="pills"
        className="sort-menu"
        justified={false}
        onSelect={[Function]}
        pullLeft={false}
        pullRight={false}
        stacked={false}
      >
        <ul
          className="sort-menu nav nav-pills"
          role={null}
        >
          <NavItem
            active={false}
            activeKey=""
            className="sort-menu-item"
            disabled={false}
            eventKey="0"
            href="#"
            key=".$0"
            onSelect={[Function]}
          >
            <li
              className="sort-menu-item"
              role="presentation"
            >
              <SafeAnchor
                componentClass="a"
                disabled={false}
                href="#"
                onClick={[Function]}
                role="button"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  role="button"
                >
                  a
                </a>
              </SafeAnchor>
            </li>
          </NavItem>
          <NavItem
            active={false}
            activeKey=""
            className="sort-menu-item"
            disabled={false}
            eventKey="1"
            href="#"
            key=".$1"
            onSelect={[Function]}
          >
            <li
              className="sort-menu-item"
              role="presentation"
            >
              <SafeAnchor
                componentClass="a"
                disabled={false}
                href="#"
                onClick={[Function]}
                role="button"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  role="button"
                >
                  b
                </a>
              </SafeAnchor>
            </li>
          </NavItem>
          <NavItem
            active={false}
            activeKey=""
            className="sort-menu-item"
            disabled={false}
            eventKey="2"
            href="#"
            key=".$2"
            onSelect={[Function]}
          >
            <li
              className="sort-menu-item"
              role="presentation"
            >
              <SafeAnchor
                componentClass="a"
                disabled={false}
                href="#"
                onClick={[Function]}
                role="button"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  role="button"
                >
                  c
                </a>
              </SafeAnchor>
            </li>
          </NavItem>
        </ul>
      </Nav>
    </span>
  </div>
</SortMenu>
`;
